multicluster/controllers/multicluster/member/serviceexport_controller.go

```diff
@@ -231,19 +237,37 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			Namespace: req.Namespace,
 		},
 	}
-	readyAddresses := 0
-	err = r.Client.Get(ctx, req.NamespacedName, eps)
-	if err == nil {
-		for _, subsets := range eps.Subsets {
-			readyAddresses = readyAddresses + len(subsets.Addresses)
+
+	var noEndpoint bool
+	if r.endpointSliceEnabled {
+		eps.Subsets, err = r.getSubsetsFromEndpointSlice(ctx, req)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if eps.Subsets == nil {
+			noEndpoint = true
 		}
 	} else {
-		if !apierrors.IsNotFound(err) {
-			klog.ErrorS(err, "Failed to get Endpoints", req.String())
-			return ctrl.Result{}, err
+		readyAddresses := 0
```

Comment:
Would you add a func like getSubnetsFromEndpoints?

Reply:
Done.